components/idp-mgt/org.wso2.carbon.idp.mgt/src/test/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAOTest.java

```diff
@@ -0,0 +1,1690 @@
+            addTestIdps(idPManagementDAO);
+
+            //forceDeleteIdP
+            assertThrows(IdentityProviderManagementException.class, () ->
+                    idPManagementDAO.forceDeleteIdP(idpName, tenantId, TENANT_DOMAIN));
+        }
+    }
+
+    @Test(dataProvider = "deleteIdPData")
+    public void testForceDeleteIdPByResourceId(String idpName, int tenantId) throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        IdPManagementDAO idPManagementDAO = new IdPManagementDAO();
+
+        try (Connection connection = getConnection(DB_NAME)) {
+            when(IdentityDatabaseUtil.getDBConnection(anyBoolean())).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+            addTestIdps(idPManagementDAO);
+
+            //forceDeleteIdPByResourceId
+            String uuid = idPManagementDAO.getIdPByName(connection, idpName, tenantId, TENANT_DOMAIN).getResourceId();
+            idPManagementDAO.forceDeleteIdPByResourceId(uuid, tenantId, TENANT_DOMAIN);
+            int resultSize = getIdPCount(connection, idpName, tenantId);
+            assertEquals(resultSize, 0, "'forceDeleteIdPByResourceId' method fails");
+        }
+    }
+
+    @Test(dataProvider = "deleteIdPExceptionData")
+    public void testForceDeleteIdPByResourceIdException(String uuid, int tenantId) throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        IdPManagementDAO idPManagementDAO = new IdPManagementDAO();
+
+        try (Connection connection = getConnection(DB_NAME)) {
+            when(IdentityDatabaseUtil.getDBConnection(anyBoolean())).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+            addTestIdps(idPManagementDAO);
+
+            //forceDeleteIdPByResourceId
+            assertThrows(IdentityProviderManagementException.class, () ->
+                    idPManagementDAO.forceDeleteIdPByResourceId(uuid, tenantId, TENANT_DOMAIN));
+        }
+    }
+
+    @DataProvider
+    public Object[][] deleteIdPsData() {
+
+        return new Object[][]{
+                {SAMPLE_TENANT_ID},
+                {SAMPLE_TENANT_ID2},
+        };
+    }
+
+    @Test(dataProvider = "deleteIdPsData")
+    public void testDeleteIdPs(int tenantId) throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        IdPManagementDAO idPManagementDAO = new IdPManagementDAO();
+
+        try (Connection connection = getConnection(DB_NAME)) {
+            when(IdentityDatabaseUtil.getDBConnection(anyBoolean())).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+
+            addTestIdps(idPManagementDAO);
+            idPManagementDAO.deleteIdPs(tenantId);
+        }
+
+        try (Connection connection = getConnection(DB_NAME)) {
+            String query = "SELECT * FROM IDP WHERE TENANT_ID=?";
```

Comment:
Don't we have existing query definitions for these queries?

Reply:
Replaced the queries with already existing queries, which has existing definitions. But keep remaining queries, which haven't existing definitions.